source/base/patterns.cc

```diff
@@ -548,7 +548,7 @@ namespace Patterns
     std::string tmp(sequence);
 
     // remove whitespace at beginning
-    while ((tmp.length() != 0) && (std::isspace(tmp[0])))
+    while ((tmp.length() != 0) && ((std::isspace(tmp[0])) != 0))
```

Comment:
This is introducing some unnecessary parentheses, but it's not worth worrying about.

Same below.

Reply:
See #13167.